checks.d/haproxy.py

```diff
@@ -303,14 +321,13 @@ def _process_backend_hosts_metric(self, hosts_statuses, services_incl_filter=Non
 
     def _process_status_metric(self, hosts_statuses, collect_status_metrics_by_host,
                                services_incl_filter=None, services_excl_filter=None,
-                               count_status_by_service=True):
-        agg_statuses = defaultdict(lambda: {'available': 0, 'unavailable': 0})
+                               collate_status_tags_per_host=False, count_status_by_service=True):
+        agg_statuses_counter = defaultdict(lambda: {status: 0 for status in Services.COLLATED_STATUSES})
 
-        # use a counter unless we have a unique tag set to gauge
-        counter = defaultdict(int)
-        if count_status_by_service and collect_status_metrics_by_host:
-            # `service` and `backend` tags will exist
-            counter = None
+        reported_statuses = Services.ALL_STATUSES
+        if collate_status_tags_per_host:
+            reported_statuses = Services.COLLATED_STATUSES
+        counter = defaultdict(lambda: {status: 0 for status in reported_statuses})
```

Comment:
Although it was already named like that before those changes, can we use a more comprehensive varaible name than `counter` ?